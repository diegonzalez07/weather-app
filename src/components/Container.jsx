import React from "react";

export const Container = ({ children }) => {
  return (
    <div className="container-fluid">
      <div className="row">
        <div className="col">
          <>{children}</>
        </div>
      </div>
    </div>
  );
};
