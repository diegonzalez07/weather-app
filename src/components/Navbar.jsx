import React from "react";

export const Navbar = () => {
  return (
    <nav className="navbar navbar-dark bg-primary mb-3 p-3">
      <h1 className="text-light">WeatherApp</h1>
    </nav>
  );
};
