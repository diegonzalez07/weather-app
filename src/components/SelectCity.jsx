import { useState } from "react";

export const SelectCity = ({ onSetCity, selectedCity }) => {
  const handleChangeCity = ({ target }) => {
    onSetCity(target.value);
  };

  return (
    <form>
      <select
        className="form-select"
        value={selectedCity}
        onChange={handleChangeCity}
      >
        <option value="miCity">Mi ciudad</option>
        <option value="bogota">Bogotá</option>
        <option value="madrid">Madrid</option>
        <option value="londres">Londres</option>
        <option value="parís">Paris</option>
        <option value="montevideo">Montevideo</option>
      </select>
    </form>
  );
};
