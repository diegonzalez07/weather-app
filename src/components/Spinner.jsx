export const Spinner = () => {
  return (
    <div className="d-flex justify-content-center">
      <div className="spinner-grow mt-4" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
      <div className="spinner-grow mt-4" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
      <div className="spinner-grow mt-4" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
      <div className="spinner-grow mt-4" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
      <div className="spinner-grow mt-4" role="status">
        <span className="visually-hidden">Loading...</span>
      </div>
    </div>
  );
};
