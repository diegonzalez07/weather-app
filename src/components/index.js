export * from "./Container";
export * from "./Navbar";
export * from "./SelectCity";
export * from "./Spinner";
export * from "./WeatherCard";
export * from "./WeatherCurrentCard";
export * from "./WeatherCurrents";
export * from "./WeatherForecastGrid";
