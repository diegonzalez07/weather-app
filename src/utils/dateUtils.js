import { format, parseISO } from "date-fns";

export const getFormatBy = (type, date) => {
  if (type === "day") {
    return format(parseISO(date), "dd-MM-yyyy");
  }

  return format(parseISO(date), "HH:mm");
};
