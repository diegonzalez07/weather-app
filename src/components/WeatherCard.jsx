import { getFormatBy } from "../utils/dateUtils";
import PropTypes from "prop-types";

export const WeatherCard = ({
  id,
  atmosphericPressure,
  hour,
  humidity,
  icon,
  temperature,
  thermalSensation,
}) => {
  return (
    <div className="card mb-3 shadow">
      <h5 className="card-header text-center">
        {`${getFormatBy("day", hour)} | ${getFormatBy("hour", hour)} hs.`}
      </h5>

      <div className="container mt-2">
        <div className="row align-items-center p-3">
          <div className="col">
            <img
              src={`http://openweathermap.org/img/w/${icon}.png`}
              className="rounded img-thumbnail"
              height="0.3rem"
              alt={`${icon}_${id}`}
            />
          </div>
          <div className="col">
            <h2 className="m-0 text-end">{temperature.toFixed(1)}°C</h2>
          </div>
        </div>
      </div>

      <div className="card-body">
        <div className="container mt-2">
          <div className="row align-items-center text-center">
            <div className="col">
              <p className="m-0">ST: {thermalSensation.toFixed(1)}°C</p>
            </div>
            <div className="col">
              <p className="m-0">Hum: {humidity}%</p>
            </div>
          </div>
          <hr />
          <div className="row align-items-center text-center">
            <div className="col">
              <p className="m-0">Pres AT: {atmosphericPressure} Pa</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

WeatherCard.propTypes = {
  id: PropTypes.number.isRequired,
  atmosphericPressure: PropTypes.number.isRequired,
  hour: PropTypes.string.isRequired,
  humidity: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.number.isRequired,
  thermalSensation: PropTypes.number.isRequired,
};
