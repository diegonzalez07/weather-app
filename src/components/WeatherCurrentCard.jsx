import React from "react";
import { capitalizeFirstLetter } from "../utils/stringUtils";

export const WeatherCurrentCard = ({ currentWeather }) => {
  const {
    atmosphericPressure,
    humidity,
    icon,
    temperature,
    thermalSensation,
    visibility,
    wind,
    id,
  } = currentWeather;
  return (
    <div className="card text-bg-info mb-3 shadow-lg">
      <div className="card-body">
        <div className="container mt-2">
          <div className="row p-3 align-items-center">
            <div className="col-11">
              <div className="row align-items-center">
                <div className="col-3">
                  <h2 className="m-0">Temp: {temperature.toFixed(1)}°C</h2>
                </div>
                <div className="col-9">
                  <div className="row align-items-center">
                    <div className="col-5">
                      <p className="m-0">
                        Sensación Térmica: {thermalSensation.toFixed(1)}°C
                      </p>
                    </div>
                    <div className="col">
                      <p className="m-0">Humedad: {humidity}%</p>
                    </div>
                    <div className="col">
                      <p className="m-0">Viento: {wind} km/h</p>
                    </div>
                  </div>
                  <hr />
                  <div className="row align-items-center">
                    <div className="col-5">
                      <p className="m-0">
                        Presión Atmosférica: {atmosphericPressure} hPa
                      </p>
                    </div>
                    <div className="col">
                      <p className="m-0">Visibilidad: {visibility} mts</p>
                    </div>
                  </div>
                </div>
              </div>
            </div>

            <div className="col-1">
              <img
                src={`http://openweathermap.org/img/w/${icon}.png`}
                className="rounded img-thumbnail"
                height="0.3rem"
                alt={`${icon}_${id}`}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
