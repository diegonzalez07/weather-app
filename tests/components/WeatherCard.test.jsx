import { render, screen } from "@testing-library/react";
import { WeatherCard } from "../../src/components/WeatherCard";
import { getFormatBy } from "../../src/utils/dateUtils";

describe("Test en <WeatherCard />", () => {
  const mockData = {
    id: 1,
    atmosphericPressure: 2,
    hour: "2022-09-12 06:00:00",
    humidity: 3,
    icon: "icon",
    temperature: 1234,
    thermalSensation: 456,
  };
  test("Debe hacer match con Snapshot", () => {
    const { container } = render(<WeatherCard {...mockData} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar imagen con url y alt indicado", () => {
    render(<WeatherCard {...mockData} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(`http://openweathermap.org/img/w/${mockData.icon}.png`);
    expect(alt).toBe(`${mockData.icon}_${mockData.id}`);
  });

  test("Debe mostrar el titulo en el componente", () => {
    render(<WeatherCard {...mockData} />);

    expect(
      screen.getByText(
        `${getFormatBy("day", mockData.hour)} | ${getFormatBy(
          "hour",
          mockData.hour
        )} hs.`
      )
    ).toBeTruthy();
  });
});
