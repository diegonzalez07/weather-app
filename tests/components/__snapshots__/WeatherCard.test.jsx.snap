// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <WeatherCard /> Debe hacer match con Snapshot 1`] = `
<div>
  <div
    class="card mb-3 shadow"
  >
    <h5
      class="card-header text-center"
    >
      12-09-2022 | 06:00 hs.
    </h5>
    <div
      class="container mt-2"
    >
      <div
        class="row align-items-center p-3"
      >
        <div
          class="col"
        >
          <img
            alt="icon_1"
            class="rounded img-thumbnail"
            height="0.3rem"
            src="http://openweathermap.org/img/w/icon.png"
          />
        </div>
        <div
          class="col"
        >
          <h2
            class="m-0 text-end"
          >
            1234.0
            °C
          </h2>
        </div>
      </div>
    </div>
    <div
      class="card-body"
    >
      <div
        class="container mt-2"
      >
        <div
          class="row align-items-center text-center"
        >
          <div
            class="col"
          >
            <p
              class="m-0"
            >
              ST: 
              456.0
              °C
            </p>
          </div>
          <div
            class="col"
          >
            <p
              class="m-0"
            >
              Hum: 
              3
              %
            </p>
          </div>
        </div>
        <hr />
        <div
          class="row align-items-center text-center"
        >
          <div
            class="col"
          >
            <p
              class="m-0"
            >
              Pres AT: 
              2
               Pa
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
